### PR TITLE
ci: build for 32 bit, libs: fix overflow

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

- build for 32 bit in ci 
- fix overflow in libs/math

Closes: #5693 

Opened this to demonstrate the changes to enable arm for releases. We can close it if we decide to not support 32-bit.

### CODE REVIEW

File: libs/math/fraction.go [hunk 1]
```diff
@@ -42,7 +42,8 @@ func ParseFraction(f string) (Fraction, error) {
 		return Fraction{}, errors.New("denominator can't be 0")
 	}
 	if numerator > math.MaxInt64 || denominator > math.MaxInt64 {
-		return Fraction{}, fmt.Errorf("value overflow, numerator and denominator must be less than %d", math.MaxInt64)
+		return Fraction{}, fmt.Errorf("value overflow, numerator and denominator must be less than %s",
+			strconv.FormatInt(math.MaxInt64, 10))
```
Comment (melekes) [inline libs/math/fraction.go:46]:
```suggestion
			math.MaxInt64)
```

does this work? I mean fmt.Errorf(... %s, math.MaxInt64)

Comment (marbar3778) [reply libs/math/fraction.go:46]: It overflows as well.

**Comment by melekes** [comment]
do we actually need to run tests on arm32? isn't it Go job to make sure arithmetic is correct. not sure what benefits it brings.. am I missing something?

**Comment by tessr** [comment]
>  isn't it Go job to make sure arithmetic is correct.

The compiler can't catch runtime overflows, though.

**Review by melekes — APPROVED** [review]

**Comment by melekes** [comment]
> > isn't it Go job to make sure arithmetic is correct.
> 
> The compiler can't catch runtime overflows, though.

right, thanks

File: .goreleaser.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@ builds:
       - windows
     goarch:
       - amd64
+      - arm
```
Comment (alessio) [inline .goreleaser.yml:20]: Why not adding 386 as well?

File: libs/math/fraction.go (hunk 1 above)
Comment (erikgrinaker) [reply libs/math/fraction.go:46]: The only reason `FormatInt()` works is that the parameter is of type `int64` so it implicitly casts the constant to an int64. It's simpler to just do `fmt.Errorf("... %s", int64(math.MaxInt64))`. 32-bit archs do support 64-bit ints, they just don't default to them.

File: .goreleaser.yml (hunk 2 above)
Comment (marbar3778) [reply .goreleaser.yml:20]: We haven't had a request for it. I am wary of adding more things to `goarch` without a request from users.

Comment (alessio) [reply .goreleaser.yml:20]: That's fair :+1:

**Pull request merged**